src/msw/anybutton.cpp

```diff
@@ -812,29 +812,29 @@ wxBitmap wxAnyButton::DoGetBitmap(State which) const
 
 void wxAnyButton::DoSetBitmap(const wxBitmapBundle& bitmapBundle, State which)
 {
+    // Normal image sets images for all states of the button, or deletes the
+    // images if the bundle is invalid.
+    // Delete the wxButtonImageData so when the new one is created, all states
+    // are initialized.
+    if (which == State_Normal)
+    {
+        delete m_imageData;
+        m_imageData = NULL;
+    }
+
     if ( !bitmapBundle.IsOk() )
     {
-        if ( m_imageData  )
+        if ( m_imageData && which != State_Normal )
```

Comment:
This doesn't seem to be necessary? I.e. if `which == State_Normal` then `m_imageData` is already null anyhow.

I'll remove this change if you don't mind.

Reply:
You're right. I just kept all the checks from the original if-statements. This one seems unnecessary now.